x-pack/filebeat/fbreceiver/receiver_test.go

```diff
@@ -380,14 +361,14 @@ func TestReceiverDegraded(t *testing.T) {
 	}
 }
 
-func genSocketPath() string {
-	randData := make([]byte, 16)
-	for i := range len(randData) {
-		randData[i] = uint8(rand.UintN(255)) //nolint:gosec // 0-255 fits in a uint8
-	}
-	socketName := base64.URLEncoding.EncodeToString(randData) + ".sock"
-	socketDir := os.TempDir()
-	return filepath.Join(socketDir, socketName)
+func genSocketPath(t *testing.T) string {
+	t.Helper()
+	socketName, err := uuid.NewV4()
```

Comment:
On Windows there is a non obvious requirement that the whole named pipe path length be less than 256 characters.

https://learn.microsoft.com/en-us/windows/win32/api/namedpipeapi/nf-namedpipeapi-createnamedpipew

is a v4 UUID guaranteed to be less than that?

Reply:
Yes, it's 36 chars including the hyphens.